stock_picking_consolidation_priority/models/stock_move.py

```diff
@@ -0,0 +1,119 @@
+        -- through stock_move_move_rel to find all the origin moves
+        origins (id, picking_id, is_consolidation_dest) AS (
+            -- Find the destination move of the current moves which have the flag
+            -- consolidate_priority on their picking type. From there, find all
+            -- the other moves of the consolidation transfer.
+            -- They are the starting point to search the origin moves.
+            -- In the final query, the rows with "consolidate_priority" equal
+            -- to true are excluded, because we want to raise the priority of
+            -- the moves *before* the picking type with the flag
+            SELECT consolidation_dest_moves.id,
+                   stock_picking.id,
+                   stock_picking_type.consolidate_priority as is_consolidation_dest
+            FROM stock_move_move_rel
+            INNER JOIN stock_move move_dest
+            ON move_dest.id = stock_move_move_rel.move_dest_id
+            INNER JOIN stock_picking
+            ON stock_picking.id = move_dest.picking_id
+            -- select *all* the moves of the transfer with the consolidation flag,
+            -- origin moves will be searched for all of them in the recursive part
+            INNER JOIN stock_move consolidation_dest_moves
+            ON consolidation_dest_moves.picking_id = stock_picking.id
+            INNER JOIN stock_picking_type
+            ON stock_picking_type.id = stock_picking.picking_type_id
+            WHERE stock_move_move_rel.move_orig_id IN %s
+            AND stock_picking_type.consolidate_priority = true
+
+            -- We use union here to keep duplicate in case a move both has the
+            -- flag "consolidate_priority" on its picking type AND is the
+            -- origin move for another move with the flag (e.g, option
+            -- activated both on PACK and OUT). Anyway, the final query filters
+            -- on the second part of the union.
+            UNION ALL
+
+            -- recurse to find all the origin moves which have a destination that
+            -- needs priority consolidation
+            SELECT move_orig.id,
+                   move_orig.picking_id,
+                   false as is_consolidation_dest
+            FROM stock_move move_orig
+            INNER JOIN stock_move_move_rel
+            ON move_orig.id = stock_move_move_rel.move_orig_id
+            INNER JOIN origins
+            ON origins.id = stock_move_move_rel.move_dest_id
+        )
+        SELECT id FROM origins WHERE is_consolidation_dest = false
+        """
+        return (query, (tuple(self.ids),))
+
+    def _consolidate_priority_domain(self):
+        return [("state", "not in", ("cancel", "done"))]
+
+    def _consolidate_priority_values(self):
+        return {"priority": self._consolidate_priority_value}
+
+    def _consolidate_priority(self):
+        query, params = self._query_get_consolidate_moves()
+        self.env.cr.execute(query, params)
```

Comment:
Shouldn't the models be flushed before executing the SQL?

Reply:
yes, thanks! pushed a fixup